src/tracked.jl

```diff
@@ -274,6 +274,7 @@ for R in REAL_TYPES
 end
 
 Base.promote_rule(::Type{R}, ::Type{TrackedReal{V,D,O}}) where {R<:Real,V,D,O} = TrackedReal{promote_type(R,V),D,O}
+Base.promote_rule(::Type{R}, ::Type{TrackedReal{V, D, O}}) where {R<:AbstractIrrational,V,D,O} = TrackedReal{promote_type(R,V),D,O}
```

Comment:
Maybe similar to Tracker (https://github.com/FluxML/Tracker.jl/blob/b26b29fc1a9e9de60f2969ba0605eebdee168215/src/lib/real.jl#L64-L65) we can only define
```suggestion
Base.promote_rule(::Type{TrackedReal{V,D,O}}, ::Type{R}) where {V,D,O,R<:Real}) = TrackedReal{promote_type(R,V),D,O}
```
? From a quick look in Julia base, generally it seems more common to make the "important" type the first argument, hence maybe this helps to avoid method ambiguity issues.